.github/workflows/release.yml

```diff
@@ -4,18 +4,20 @@ name: Check and realese cosmpy
 # if package version is newer it build, upload, tag and make a release page on github
 # secrets PYPI_USERNAME and PYPI_PASSWORD are required!
 on:
-  push:
-    branches: [ master ]
+  pull_request:
+    types:
+      - opened
```

Comment:
You need to keep the master filter (release checks go into master)
Also remove types: opened (this keeps it to the default of reopen, sync and open)